src/predis/mock.clj

```diff
@@ -287,6 +290,19 @@
       (when (<= idx last-idx)
         (get vs idx'))))
 
+  (core/linsert [this k pos pivot v]
+    (assert (or (= pos "before") (= pos "after")) err-syntax)
+    (let [vs (vec (core/get this k))]
+      (if (not (seq vs))
+        0
+        (if (contains? (set vs) pivot)
+          (let [[before after] (split-at ((if (= pos "before") util/id inc)
+                                            (.indexOf vs pivot)) vs)
```

Comment:
Double parens are a red flag to me - in this case dynamically selecting a function and then invoking it seems like more confusion that it's worth. I'd prefer a simpler solution just conditionally combining `inc` and `indexOf`